locations/microdata_parser.py

```diff
@@ -13,7 +13,7 @@ def token_split(val):
 
 
 def top_level_items(selector: parsel.Selector):
-    yield from selector.xpath("//*[@itemscope][not(@itemprop)]")
+    yield from selector.xpath("//*[@itemscope]")
```

Comment:
I think this is going to have ramifications for the other spiders. I can't test it now, but I'll probably have time tomorrow

Reply:
It was incorrect per the HTML specification that allows itemprop and itemscope to be used together on the same element. There are examples of such at https://html.spec.whatwg.org/#the-basic-syntax

Reply:
Yeah but this is a property of another itemscope.

Reply:
The specific top-level element I wanted to catch with this change was:

```<div itemprop="location" itemscope itemtype="http://schema.org/Store">```

(from example https://www.qbd.com.au/locations/australia-fair/)

§5.2.2 of the standard states:
```**Every** [HTML element](https://html.spec.whatwg.org/#html-elements) may have an itemscope attribute specified. The [itemscope](https://html.spec.whatwg.org/#attr-itemscope) attribute is a [boolean attribute](https://html.spec.whatwg.org/#boolean-attribute).```

§5.2.3 of the standard then states:
```**Every** [HTML element](https://html.spec.whatwg.org/#html-elements) may have an [itemprop](https://html.spec.whatwg.org/#names:-the-itemprop-attribute) attribute specified, if doing so [adds one or more properties](https://html.spec.whatwg.org/#the-properties-of-an-item) to one or more [items](https://html.spec.whatwg.org/#concept-item) (as defined below).```

§5.2.4 of the standard then states:
```The property value of a name-value pair added by an element with an [itemprop](https://html.spec.whatwg.org/#names:-the-itemprop-attribute) attribute is as given for the first matching case in the following list:
If the **element also has an [itemscope](https://html.spec.whatwg.org/#attr-itemscope) attribute**```

Reply:
I know it's valid for a itemprop to also be a itemscope. That's how we get "nested dicts" in Microdata.

I thought it was invalid for that to happen on the top objects. (What is location a prop of?) I may be wrong, it might be valid. But I still think it will cause issues with our Microdata parser. With more complicated Microdata with more than one parent and children objects.

Reply:
I'll find an example of "complicated Microdata" tomorrow and test it

Reply:
Yeah, so it doesn't even need to be complicated. Any "nested dict" would now be extracted multiple times, eg:

```python
{'@context': 'https://schema.org',
 '@graph': [{'@type': 'Store',
             'name': 'QBD Books - Australia Fair',
             'priceRange': '$1 - $500',
             'image': 'https://www.qbd.com.au/img/assets/c45828494175487845208af0f97397f5.jpg',
             'address': {'@type': 'PostalAddress',
                         'streetAddress': 'Shop G091A, Australia Fair Shopping '
                                          'Centre 42 Marine Parade',
                         'addressLocality': 'Southport',
                         'addressRegion': 'QLD',
                         'postalCode': '4215'},
             'geo': {'@type': 'GeoCoordinates',
                     'latitude': '-27.968185',
                     'longitude': '153.415649'},
             'openingHours': ['Mo 8:30-17:30',
                              'Tu 8:30-17:30',
                              'We 8:30-17:30',
                              'Th 8:30-21:00',
                              'Fr 8:30-17:30',
                              'Sa 8:30-17:30',
                              'Su 10:00-16:00'],
             'telephone': '(07) 5528 0288',
             'email': 'mailto:australiafair@qbd.com.au'},
            {'@type': 'PostalAddress',
             'streetAddress': 'Shop G091A, Australia Fair Shopping Centre 42 '
                              'Marine Parade',
             'addressLocality': 'Southport',
             'addressRegion': 'QLD',
             'postalCode': '4215'},
            {'@type': 'GeoCoordinates',
             'latitude': '-27.968185',
             'longitude': '153.415649'}]}
```

We want:
```python
{'@context': 'https://schema.org',
 '@graph': [{'@type': 'Store',
             'name': 'QBD Books - Australia Fair',
             'priceRange': '$1 - $500',
             'image': 'https://www.qbd.com.au/img/assets/c45828494175487845208af0f97397f5.jpg',
             'address': {'@type': 'PostalAddress',
                         'streetAddress': 'Shop G091A, Australia Fair Shopping '
                                          'Centre 42 Marine Parade',
                         'addressLocality': 'Southport',
                         'addressRegion': 'QLD',
                         'postalCode': '4215'},
             'geo': {'@type': 'GeoCoordinates',
                     'latitude': '-27.968185',
                     'longitude': '153.415649'},
             'openingHours': ['Mo 8:30-17:30',
                              'Tu 8:30-17:30',
                              'We 8:30-17:30',
                              'Th 8:30-21:00',
                              'Fr 8:30-17:30',
                              'Sa 8:30-17:30',
                              'Su 10:00-16:00'],
             'telephone': '(07) 5528 0288',
             'email': 'mailto:australiafair@qbd.com.au'}]}
```

I'm pushing a change to override `extract_microdata`, I think ideally we'd just want to override `top_level_items`, but I can't immediately see how, and I want to swim again today. Let me know what you think.